src/app/shared/services/courses/courses.service.ts

```diff
@@ -19,17 +20,21 @@ export class CoursesService implements CoursesServiceInterface {
   endPoint = `${environment.restEndPoint}/courses`;
   courses: Course[];
 
-  constructor(private http: HttpClient) {}
+  constructor(private http: HttpClient,
+    private spinner: NgxSpinnerService
+  ) {}
 
   find(searchString: string, start: number, countItems: number = 20): Observable<CoursesResponse> {
     let url = `${this.endPoint}?search=${searchString}&start=${start}&count=${countItems}`;
 
+    this.spinner.show();
```

Comment:
According to **point 7** in the task list, spinner should appear on **every** API call.
Maybe it's better to change his state within interceptor.